build-test.sh

```diff
@@ -152,7 +152,7 @@ generate_layout()
     # ===
     # =========================================================================================
 
-    build_Tests_internal "Restore_Packages" "${__ProjectDir}/tests/build.proj" "Restore product binaries (build tests)" "-BatchRestorePackages"
+    build_MSBuild_projects "Restore_Packages" "${__ProjectDir}/tests/build.proj" "Restore product binaries (build tests)" "-BatchRestorePackages"
```

Comment:
After checking you PR I have noticed that default settings in config.json for invoking `msbuild` on unixes do not set `maxcpucount` parameter what makes all calls non parallel and slows down every `msbuild` based operation. 

Another problem is related to node reuse. AFAIR node reuse was off by default on Linux, however, this could have changed now. In such case we would need `-nodereuse=false` `msbuild` parameter. @rainersigwald may know more about current state of node reuse support on Linux.

Reply:
`dotnet build` and `dotnet msbuild` unconditionally insert `-maxcpucount` into MSBuild's command line.

Node reuse is on by default on all platforms since .NET Core SDK 2.1.300/MSBuild 15.7. Since there are many individual invocations of MSBuild here, I would expect it to provide a performance improvement and wouldn't recommend leaving it off entirely unless there's a pressing need. It might be a good idea to do a `dotnet build-server shutdown` at the end of the script (even on error paths), depending on the use case.

Reply:
I didn't change _any_ MSBuild logic in this review. That is definitely something we should investigate going forward though.